cmake/SetupDoxygen.cmake

```diff
@@ -14,17 +14,23 @@ if (DOXYGEN_FOUND)
   # Configure file that contains doxygen groups
   configure_file(docs/GroupDefs.hpp ${SPECTRE_DOXYGEN_GROUPS})
 
+  # Construct the command that calls Doxygen
   set(
     GENERATE_DOCS_COMMAND
-    ${DOXYGEN_EXECUTABLE}
-    ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml
+    "${DOXYGEN_EXECUTABLE} ${PROJECT_BINARY_DIR}/docs/DoxyfileHtml"
   )
-  # Only outputs warnings with a few lines of context to stderr. Fails with exit
-  # code 1 if warnings are found. Remains silent and succeeds with exit code 0
-  # if everything is fine.
+  # Construct the command that calls doxygen, but only outputs warnings with a
+  # few lines of context to stderr. Fails with exit code 1 if warnings are
+  # found. Remains silent and succeeds with exit code 0 if everything is fine.
+  # We write this into a shell script to make it easier to call in
+  # `add_custom_target`.
```

Comment:
The reasoning here seems incorrect. I thought this is done for easier control of the post-processing step?